standards/application/waku-api.md

```diff
@@ -0,0 +1,327 @@
+      cluster_id:
+        type: uint
+      auto_sharding_config:
+        type: AutoShardingConfig
+        default: DefaultAutoShardingConfig
+        description: "The auto-sharding config, if sharding mode is `auto`"
+      message_validation:
+        type: MessageValidation
+        # If the default config for TWN is not used, then we still provide a message validation default 
+        default: DefaultMessageValidation
+
+  NetworkingConfig:
+    type: string
+    fields:
+      listen_ipv4:
+        # Is not applicable in some environments such as browser.
+        type: string
+        default: "0.0.0.0"
+        description: "The network IP address on which libp2p and discv5 listen for inbound connections" 
+      p2p_tcp_port:
+        # Is not applicable in non-TCP environments such as browser
+        type: uint
+        default: 60000
+        description: "The TCP port used for libp2p, relay, aka, general p2p message routing."
+      discv5_udp_port:
+        # Is not applicable in non-UDP environments such as browser
+        type: uint
+        default: 9000
+        description: "The UDP port used for discv5."
+
+  AutoShardingConfig:
+    type: struct
+    fields:
+      num_shards_in_cluster:
+        type: uint
+        description: "The number of shards in the configured cluster; this is a globally agreed value for each cluster."
+
+  MessageValidation:
+    type: struct
+    fields:
+      max_message_size:
+        type: string
+        default: "150 KiB"
+        description: "Maximum message size. Accepted units: KiB, KB, and B. e.g. 1024KiB; 1500 B; etc."
+      # For now, RLN is the only message validation available
+      rln_config:
+        type: RlnConfig
+        # If the default config for TWN is not used, then we do not apply RLN
+        default: none
+
+  RlnConfig:
+    type: struct
+    fields:
+      contract_address:
+        type: string
+        description: "The address of the RLN contract exposes `root` and `getMerkleRoot` ABIs"
+      chain_id:
+        type: uint
+        description: "The chain id on which the RLN contract is deployed"
+      epoch_size_sec:
+        type: uint
+        description: "The epoch size to use for RLN, in seconds"
+```
+
+#### Function definitions
+
+```yaml
+functions:
+  createNode:
+    description: "Initialise a Waku node instance"
+    parameters:
+      - name: config
+        type: Config
+        description: "The Waku node configuration."
+    returns:
+        type: result<WakuNode, error>
+```
+
+#### Predefined values
+
+```yaml
+values:
+
+  DefaultNetworkingConfig:
+    type: NetworkConfig
+    fields:
+      listen_address: "0.0.0.0"
+      p2p_tcp_port: 60000
+      discv5_udp_port: 9000
+
+  TheWakuNetworkPreset:
+    type: WakuConfig
+    fields:
+      remote_nodes: [ "enrtree://AIRVQ5DDA4FFWLRBCHJWUWOO6X6S4ZTZ5B667LQ6AJU6PEYDLRD5O@sandbox.waku.nodes.status.im" ]
```

Comment:
I am not sure it worth mentioning `remote nodes` and `contract address` as they can get outdated rather fast: unpredicted contract changes like we did recently etc.